### PR TITLE
ADD: Global Message UI

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/6ej2jcn46a6uaq3upme1u5vg14

[download apk](http://filestorage.bluewallet.io:1488/1597192075.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/zca56kfr6v5u3q07dh88htyrm0

[download apk](http://filestorage.bluewallet.io:1488/1597251792.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/n3vccm7aftgyw79urudqrq29hw

[download apk](http://filestorage.bluewallet.io:1488/1597275512.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/5ynh39xbgg9y4x1r9zhen2vuj0

[download apk](http://filestorage.bluewallet.io:1488/1597278483.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/hb0908v01149gpcv4r2eadyzf8

[download apk](http://filestorage.bluewallet.io:1488/1597933918.apk)

**Comment by Overtorment** [comment]
screenshots in this kind of PRs would be helpful

File: components/BlueGlobalMessageContext.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import React, { createContext, useState, useEffect, useRef, useCallback } from 'react';
+
+const initialMessage = {
+  message: '',
+  type: null,
+  visible: false,
+  dismissable: true,
```
Comment (Overtorment) [inline components/BlueGlobalMessageContext.js:7]: dont see, is there an autohide property somewhere? bubbles should hide automatically by default

File: components/BlueGlobalMessageContext.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import React, { createContext, useState, useEffect, useRef, useCallback } from 'react';
+
+const initialMessage = {
+  message: '',
+  type: null,
```
Comment (Overtorment) [inline components/BlueGlobalMessageContext.js:5]: types should be described somewhere. basically, we need error, warning, info, success

**Review by Overtorment — CHANGES_REQUESTED** [review]
needs more work

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/y5zdmknp5kjtryhrhqf7v27ucm

[download apk](http://filestorage.bluewallet.io:1488/1598210707.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/qvwg927agg5n2qbhy9k8cw7ygr

[download apk](http://filestorage.bluewallet.io:1488/1598224953.apk)

File: components/BlueGlobalMessageContext.js (hunk 1 above)
Comment (marcosrdz) [reply components/BlueGlobalMessageContext.js:14]: they do. lines 36         timeout.current = setTimeout(hide, 3000);

File: components/BlueGlobalMessage.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import { useTheme } from '@react-navigation/native';
+import loc from '../loc';
+
+/*
+
+How to use: 
+
+Class Component Example:
+    WalletsList.contextType = BlueGlobalMessageContext;
+    this.context.show({ message: "Importing Wallet..."});
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply components/BlueGlobalMessage.js:11]: EV is needed for non-React functions. non React functions are unable to read React Context

File: class/wallet-import.js [hunk 4]
```diff
@@ -54,13 +55,15 @@ export default class WalletImport {
         A(A.ENUM.CREATED_WALLET);
         alert(loc.wallets.import_success);
         notifications.majorTomToGroundControl(w.getAllExternalAddresses(), [], []);
+        EV(EV.enum.GLOBAL_MESSAGES_HIDE);
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply class/wallet-import.js]: Hide is there for messages that shouldnt autohide..such as wallet import

File: class/wallet-import.js [hunk 5]
```diff
@@ -38,6 +38,7 @@ export default class WalletImport {
       if (wallet) {
         alert('This wallet has been previously imported.');
         WalletImport.removePlaceholderWallet();
+        EV(EV.enum.GLOBAL_MESSAGES_HIDE);
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply class/wallet-import.js]: Hide is for messages that shouldnt autohide.

**Comment by marcosrdz** [comment]
IMAGE 2020-08-23 19:32:34
IMAGE 2020-08-23 19:32:37

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/b3h9d41k4z11pzmehcu3620t3c

[download apk](http://filestorage.bluewallet.io:1488/1598226923.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ttg1tztxv5wrzj298wwzmpe830

[download apk](http://filestorage.bluewallet.io:1488/1598237695.apk)

**Comment by Overtorment** [comment]
e2e failed. probably because tests expect alerts with OK after import. maybe easier to leave them

File: components/BlueGlobalMessage.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Overtorment) [reply components/BlueGlobalMessage.js:11]: which ones are those? example?

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/qx8u071z9ycb5tvp6nxmv9hpng

[download apk](http://filestorage.bluewallet.io:1488/1598653440.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/6k7a4n7r8rtk28xz2g7y30vpmm

[download apk](http://filestorage.bluewallet.io:1488/1598667338.apk)

**Comment by marcosrdz** [comment]
> e2e failed. probably because tests expect alerts with OK after import. maybe easier to leave them

can something be done so it doesnt look for the alert?

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ztx8c9vtaxxuvz0yau5da5nnd8

[download apk](http://filestorage.bluewallet.io:1488/1599271012.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/94zrj160unexradwe5whh392x8

[download apk](http://filestorage.bluewallet.io:1488/1603764437.apk)

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by Overtorment** [comment]
im thinking about picking up this PR and implementing message bubbles UI myself, thoughts?

**Comment by marcosrdz** [comment]
Should be drastically easier now that we implemented cause context

**Comment by marcosrdz** [comment]
This code was written a very long time ago. Before the implementation of useContext. It will definitely be far easier to reintroduce now and have it talk to any part of the app without having to pass props all over

**Comment by Overtorment** [comment]
How do we do that?

**Comment by marcosrdz** [comment]
I can add it to my todo list if you'd like. Should be fairly simple by following:

https://reactjs.org/docs/context.html#consuming-multiple-contexts

**Pull request closed without merging**